outlyerapp/integrations
# [logrotate-integration] Creates logrotate integration

## Changelog
- Creates logrotate plugin that reads the logrotate state file to figure out which log files logrotate is managing as well as the last rotation datetime (then it calculates the delta with `now()`). Besides, it collects the log files sizes.
- Creates a default alert that will trigger if any of the log files don't rotate after 1 day.
- Creates integration docs, package and resources.

Below is how the dashboard looks like:

<img width="1146" alt="logrotate-dashboard" src="https://user-images.githubusercontent.com/5725856/45236184-297bda80-b2a9-11e8-8d09-32036de97272.png">

## Tests
If you want to test it, follow these steps:

1. Clone my Apache2 vagrant repository: `git clone https://github.com/jorgeacetozi/vagrant-apache2.git`
2. Set up the VM with `vagrant up`
3. ssh the VM with `vagrant ssh` and  install the Ubuntu/Debian agent
4. Force an initial rotation: `sudo logrotate -f /etc/logrotate.conf`
5. Create the Logrotate dashboard, alert and check in Outlyer
